internal/gcs/process.go

```diff
@@ -174,7 +229,17 @@ func (p *Process) CloseStderr(ctx context.Context) (err error) {
 		trace.StringAttribute("cid", p.cid),
 		trace.Int64Attribute("pid", int64(p.id)))
 
-	return p.stderr.Close()
+	if p.gcClosed() {
+		return hcs.ErrAlreadyClosed
+	}
```

Comment:
Why should we be checking if `gcClosed` in functions that don't reference `p.gc`? If it's meant to indicate that we've already closed stdio, I think maybe we should use a run once or something instead.

Reply:
The original code used `gc == nil` to indicate the process itself was closed, along with all the stdio.
We could put a separate RWMutex for the IO, and then lock the IO and bool indicating weather all IO was closed, or just skip that and check if the individual IO channels are `nil`, if that reads cleaner.